openmc/mgxs/groups.py

```diff
@@ -0,0 +1,275 @@
+from collections import Iterable
+from numbers import Real, Integral
+import copy
+import sys
+
+import numpy as np
+
+import openmc.checkvalue as cv
+
+
+if sys.version_info[0] >= 3:
+    basestring = str
+
+
+class EnergyGroups(object):
+    """An energy groups structure used for multi-group cross-sections.
+
+    Parameters
+    ----------
+    group_edges : ndarray
+        The energy group boundaries [MeV]
+    num_groups : Integral
+        The number of energy groups
+
+    Attributes
+    ----------
+    group_edges : ndarray
+        The energy group boundaries [MeV]
+    num_groups : Integral
+        The number of energy groups
+
+    """
+
+    def __init__(self, group_edges=None, num_groups=None):
+        self._group_edges = None
+        self._num_groups = None
+
+        if group_edges is not None:
+            self.group_edges = group_edges
+        if num_groups is not None:
+            self.num_groups = num_groups
```

Comment:
This doesn't work when num_groups is passed (AttributeError: can't set attribute). Might be a moot point if you take my suggestion and don't store _num_groups at all.